Backpack-SwiftUI/MapMarker/Classes/BPKPriceMapMarker.swift

```diff
@@ -0,0 +1,120 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKPriceMapMarker: View {
+    public enum State {
+        case `default`
+        case focused
+        case viewed
+        case disabled
+        
+        var fontStyle: BPKFontStyle {
+            switch self {
+            case .default:
+                return .label2
+            case .focused:
+                return .label1
+            case .viewed:
+                return .label2
+            case .disabled:
+                return .label2
+            }
+        }
+        
+        var foregroundColor: BPKColor {
+            switch self {
+            case .default:
+                return .textPrimaryInverseColor
+            case .focused:
+                return .coreAccentColor
+            case .viewed:
+                return .init(value: .white.withAlphaComponent(0.8))
+                    .darkVariant(.init(value: .black.withAlphaComponent(0.8)))
```

Comment:
nit: Should we introduce a custom color token for this use case?

Reply:
I thought the same, but checked on android and they created it on the spot. but maybe we should create it in foundations but keep private to bpk

Reply:
We can give it the same treatment as we did for buttons yea. 
A separate PR to fix this would be fine :D

Reply:
created a foundations token 👍